// @vonage/voice/dist/enums/CallStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallStatus = void 0;
/**
 * Enum representing the status of a call, including its various stages and outcomes.
 */
var CallStatus;
(function (CallStatus) {
    /**
     * The call has started.
     */
    CallStatus["STARTED"] = "started";
    /**
     * The call is ringing.
     */
    CallStatus["RINGING"] = "ringing";
    /**
     * The call has been answered.
     */
    CallStatus["ANSWERED"] = "answered";
    /**
     * The call has been detected as a machine.
     */
    CallStatus["MACHINE"] = "machine";
    /**
     * The call has been completed.
     */
    CallStatus["COMPLETED"] = "completed";
    /**
     * The call is busy.
     */
    CallStatus["BUSY"] = "busy";
    /**
     * The call has been cancelled.
     */
    CallStatus["CANCELLED"] = "cancelled";
    /**
     * The call has failed.
     */
    CallStatus["FAILED"] = "failed";
    /**
     * The call has been rejected.
     */
    CallStatus["REJECTED"] = "rejected";
    /**
     * The call has timed out.
     */
    CallStatus["TIMEOUT"] = "timeout";
    /**
     * The call went unanswered.
     */
    CallStatus["UNANSWERED"] = "unanswered";
})(CallStatus || (exports.CallStatus = CallStatus = {}));
