// @vonage/voice/dist/enums/NCCOActions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NCCOActions = void 0;
/**
 * Enum representing the available actions in a Nexmo Call Control Object (NCCO).
 */
var NCCOActions;
(function (NCCOActions) {
    /**
     * Connect the call to an endpoint or multiple endpoints.
     */
    NCCOActions["CONNECT"] = "connect";
    /**
     * Start a conversation that can include multiple participants.
     */
    NCCOActions["CONVERSATION"] = "conversation";
    /**
     * Collect input from the caller, including DTMF tones or speech.
     */
    NCCOActions["INPUT"] = "input";
    /**
     * Send a notification or message to a specified endpoint.
     */
    NCCOActions["NOTIFY"] = "notify";
    /**
     * Record the audio of a call.
     */
    NCCOActions["RECORD"] = "record";
    /**
     * Stream audio to a call.
     */
    NCCOActions["STREAM"] = "stream";
    /**
     * Play a text-to-speech message to the caller.
     */
    NCCOActions["TALK"] = "talk";
})(NCCOActions || (exports.NCCOActions = NCCOActions = {}));
