// @vonage/voice/dist/enums/TTSVoices.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TTSVoices = void 0;
/**
 * Enum representing supported Text-to-Speech (TTS) voices.
 */
var TTSVoices;
(function (TTSVoices) {
    /**
     * Aditi - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["ADITI"] = "Aditi";
    /**
     * Agnieszka - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["AGNIESZKA"] = "Agnieszka";
    /**
     * Alva - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["ALVA"] = "Alva";
    /**
     * Amy - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["AMY"] = "Amy";
    /**
     * Astrid - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["ASTRID"] = "Astrid";
    /**
     * Bianca - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["BIANCA"] = "Bianca";
    /**
     * Brian - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["BRIAN"] = "Brian";
    /**
     * Carla - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CARLA"] = "Carla";
    /**
     * Carmen - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CARMEN"] = "Carmen";
    /**
     * Carmit - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CARMIT"] = "Carmit";
    /**
     * Catarina - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CATARINA"] = "Catarina";
    /**
     * Celine - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CELINE"] = "Celine";
    /**
     * Cem - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CEM"] = "Cem";
    /**
     * Chantal - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CHANTAL"] = "Chantal";
    /**
     * Chipmunk - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CHIPMUNK"] = "Chipmunk";
    /**
     * Conchita - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CONCHITA"] = "Conchita";
    /**
     * Cristiano - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["CRISTIANO"] = "Cristiano";
    /**
     * Damayanti - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["DAMAYANTI"] = "Damayanti";
    /**
     * Dora - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["DORA"] = "Dora";
    /**
     * Emma - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["EMMA"] = "Emma";
    /**
     * Empar - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["EMPAR"] = "Empar";
    /**
     * Enrique - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["ENRIQUE"] = "Enrique";
    /**
     * Eric - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["ERIC"] = "Eric";
    /**
     * Ewa - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["EWA"] = "Ewa";
    /**
     * Felipe - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["FELIPE"] = "Felipe";
    /**
     * Filiz - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["FILIZ"] = "Filiz";
    /**
     * Geraint - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["GERAINT"] = "Geraint";
    /**
     * Giorgio - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["GIORGIO"] = "Giorgio";
    /**
     * Gwyneth - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["GWYNETH"] = "Gwyneth";
    /**
     * Hans - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["HANS"] = "Hans";
    /**
     * Henrik - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["HENRIK"] = "Henrik";
    /**
     * Ines - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["INES"] = "Ines";
    /**
     * Ioana - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["IOANA"] = "Ioana";
    /**
     * Iveta - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["IVETA"] = "Iveta";
    /**
     * Ivy - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["IVY"] = "Ivy";
    /**
     * Jacek - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["JACEK"] = "Jacek";
    /**
     * Jan - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["JAN"] = "Jan";
    /**
     * Jennifer - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["JENNIFER"] = "Jennifer";
    /**
     * Joana - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["JOANA"] = "Joana";
    /**
     * Joanna - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["JOANNA"] = "Joanna";
    /**
     * Joey - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["JOEY"] = "Joey";
    /**
     * Jordi - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["JORDI"] = "Jordi";
    /**
     * Justin - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["JUSTIN"] = "Justin";
    /**
     * Kanya - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["KANYA"] = "Kanya";
    /**
     * Karl - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["KARL"] = "Karl";
    /**
     * Kendra - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["KENDRA"] = "Kendra";
    /**
     * Kimberly - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["KIMBERLY"] = "Kimberly";
    /**
     * Laila - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["LAILA"] = "Laila";
    /**
     * Laura - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["LAURA"] = "Laura";
    /**
     * Lea - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["LEA"] = "Lea";
    /**
     * Lekha - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["LEKHA"] = "Lekha";
    /**
     * Liv - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["LIV"] = "Liv";
    /**
     * Lotte - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["LOTTE"] = "Lotte";
    /**
     * Lucia - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["LUCIA"] = "Lucia";
    /**
     * Luciana - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["LUCIANA"] = "Luciana";
    /**
     * Mads - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MADS"] = "Mads";
    /**
     * Maged - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MAGED"] = "Maged";
    /**
     * Maja - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MAJA"] = "Maja";
    /**
     * Mariska - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MARISKA"] = "Mariska";
    /**
     * Marlene - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MARLENE"] = "Marlene";
    /**
     * Mathieu - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MATHIEU"] = "Mathieu";
    /**
     * Matthew - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MATTHEW"] = "Matthew";
    /**
     * Maxim - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MAXIM"] = "Maxim";
    /**
     * Mei-Jia - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MEI_JIA"] = "Mei-Jia";
    /**
     * Melina - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MELINA"] = "Melina";
    /**
     * Mia - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MIA"] = "Mia";
    /**
     * Miguel - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MIGUEL"] = "Miguel";
    /**
     * Miren - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MIREN"] = "Miren";
    /**
     * Mizuki - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MIZUKI"] = "Mizuki";
    /**
     * Montserrat - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["MONTSERRAT"] = "Montserrat";
    /**
     * Naja - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["NAJA"] = "Naja";
    /**
     * Nicole - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["NICOLE"] = "Nicole";
    /**
     * Nikos - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["NIKOS"] = "Nikos";
    /**
     * Nora - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["NORA"] = "Nora";
    /**
     * Oskar - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["OSKAR"] = "Oskar";
    /**
     * Penelope - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["PENELOPE"] = "Penelope";
    /**
     * Raveena - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["RAVEENA"] = "Raveena";
    /**
     * Ricardo - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["RICARDO"] = "Ricardo";
    /**
     * Ruben - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["RUBEN"] = "Ruben";
    /**
     * Russell - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["RUSSELL"] = "Russell";
    /**
     * Salli - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["SALLI"] = "Salli";
    /**
     * Satu - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["SATU"] = "Satu";
    /**
     * Seoyeon - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["SEOYEON"] = "Seoyeon";
    /**
     * Sin-Ji - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["SIN_JI"] = "Sin-Ji";
    /**
     * Sora - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["SORA"] = "Sora";
    /**
     * Takumi - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["TAKUMI"] = "Takumi";
    /**
     * Tarik - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["TARIK"] = "Tarik";
    /**
     * Tatyana - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["TATYANA"] = "Tatyana";
    /**
     * Tessa - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["TESSA"] = "Tessa";
    /**
     * Tian-Tian - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["TIAN_TIAN"] = "Tian-Tian";
    /**
     * Vicki - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["VICKI"] = "Vicki";
    /**
     * Vitoria - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["VITORIA"] = "Vitoria";
    /**
     * Yelda - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["YELDA"] = "Yelda";
    /**
     * Zeina - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["ZEINA"] = "Zeina";
    /**
     * Zhiyu - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["ZHIYU"] = "Zhiyu";
    /**
     * Zuzana - Supported Text-to-Speech (TTS) voice.
     */
    TTSVoices["ZUZANA"] = "Zuzana";
})(TTSVoices || (exports.TTSVoices = TTSVoices = {}));
