// @vonage/verify/dist/enums/CheckStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckStatus = void 0;
/**
 * Enum representing the status of a verification check.
 */
var CheckStatus;
(function (CheckStatus) {
    /**
     * The verification check was successful.
     */
    CheckStatus["SUCCESS"] = "0";
    /**
     * The verification request was throttled.
     */
    CheckStatus["THROTTLED"] = "1";
    /**
     * A required parameter was missing in the verification request.
     */
    CheckStatus["MISSING_REQUIRED_PARAM"] = "2";
    /**
     * An invalid parameter was provided in the verification request.
     */
    CheckStatus["INVALID_PARAM"] = "3";
    /**
     * Invalid credentials were used in the verification request.
     */
    CheckStatus["INVALID_CREDENTIALS"] = "4";
    /**
     * An internal error occurred during the verification process.
     */
    CheckStatus["INTERNAL_ERROR"] = "5";
    /**
     * Failed to process the verification request.
     */
    CheckStatus["FAILED_TO_PROCESS"] = "6";
    /**
     * The API key used in the verification request is barred.
     */
    CheckStatus["BARRED_API_KEY"] = "8";
    /**
     * Partner quota for verifications exceeded.
     */
    CheckStatus["PARTNER_QUOTA_EXCEEDED"] = "9";
    /**
     * Concurrent verifications not allowed.
     */
    CheckStatus["CONCURRENT_VERIFICATIONS"] = "10";
    /**
     * The network is not supported for verification.
     */
    CheckStatus["UNSUPPORTED_NETWORK"] = "15";
    /**
     * Code mismatch in the verification request.
     */
    CheckStatus["CODE_MISMATCH"] = "16";
    /**
     * Invalid verification code provided.
     */
    CheckStatus["INVALID_CODE"] = "17";
    /**
     * Delivery of the verification failed.
     */
    CheckStatus["DELIVERY_FAILED"] = "19";
    /**
     * PIN not supported for verification.
     */
    CheckStatus["PIN_NOT_SUPPORTED"] = "20";
    /**
     * Destination not permitted for verification.
     */
    CheckStatus["NON_PERMITTED_DESTINATION"] = "29";
})(CheckStatus || (exports.CheckStatus = CheckStatus = {}));
