// @vonage/verify/dist/enums/VerifyLanguages.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifyLanguages = void 0;
/**
 * Enum representing supported languages for Verify requests.
 */
var VerifyLanguages;
(function (VerifyLanguages) {
    /**
     * Arabic language (ar-xa).
     */
    VerifyLanguages["ARABIC"] = "ar-xa";
    /**
     * Czech language (cs-cz).
     */
    VerifyLanguages["CZECH"] = "cs-cz";
    /**
     * Welsh language (cy-cy).
     */
    VerifyLanguages["WELSH"] = "cy-cy";
    /**
     * Welsh language, United Kingdom (cy-gb).
     */
    VerifyLanguages["WELSH_UK"] = "cy-gb";
    /**
     * Danish language (da-dk).
     */
    VerifyLanguages["DANISH"] = "da-dk";
    /**
     * German language (de-de).
     */
    VerifyLanguages["GERMAN"] = "de-de";
    /**
     * Greek language (el-gr).
     */
    VerifyLanguages["GREEK"] = "el-gr";
    /**
     * English language, Australia (en-au).
     */
    VerifyLanguages["ENGLISH_AUSTRALIA"] = "en-au";
    /**
     * English language, United Kingdom (en-gb).
     */
    VerifyLanguages["ENGLISH_UK"] = "en-gb";
    /**
     * English language, India (en-in).
     */
    VerifyLanguages["ENGLISH_INDIA"] = "en-in";
    /**
     * English language, United States (en-us).
     */
    VerifyLanguages["ENGLISH_US"] = "en-us";
    /**
     * Spanish language, Spain (es-es).
     */
    VerifyLanguages["SPANISH_SPAIN"] = "es-es";
    /**
     * Spanish language, Mexico (es-mx).
     */
    VerifyLanguages["SPANISH_MEXICO"] = "es-mx";
    /**
     * Spanish language, United States (es-us).
     */
    VerifyLanguages["SPANISH_US"] = "es-us";
    /**
     * Finnish language (fi-fi).
     */
    VerifyLanguages["FINNISH"] = "fi-fi";
    /**
     * Filipino language (fil-ph).
     */
    VerifyLanguages["FILIPINO"] = "fil-ph";
    /**
     * French language, Canada (fr-ca).
     */
    VerifyLanguages["FRENCH_CANADA"] = "fr-ca";
    /**
     * French language, France (fr-fr).
     */
    VerifyLanguages["FRENCH_FRANCE"] = "fr-fr";
    /**
     * Hindi language (hi-in).
     */
    VerifyLanguages["HINDI"] = "hi-in";
    /**
     * Hungarian language (hu-hu).
     */
    VerifyLanguages["HUNGARIAN"] = "hu-hu";
    /**
     * Indonesian language (id-id).
     */
    VerifyLanguages["INDONESIAN"] = "id-id";
    /**
     * Icelandic language (is-is).
     */
    VerifyLanguages["ICELANDIC"] = "is-is";
    /**
     * Italian language (it-it).
     */
    VerifyLanguages["ITALIAN"] = "it-it";
    /**
     * Japanese language (ja-jp).
     */
    VerifyLanguages["JAPANESE"] = "ja-jp";
    /**
     * Korean language (ko-kr).
     */
    VerifyLanguages["KOREAN"] = "ko-kr";
    /**
     * Norwegian language (nb-no).
     */
    VerifyLanguages["NORWEGIAN"] = "nb-no";
    /**
     * Dutch language (nl-nl).
     */
    VerifyLanguages["DUTCH"] = "nl-nl";
    /**
     * Polish language (pl-pl).
     */
    VerifyLanguages["POLISH"] = "pl-pl";
    /**
     * Portuguese language, Brazil (pt-br).
     */
    VerifyLanguages["PORTUGUESE_BRAZIL"] = "pt-br";
    /**
     * Portuguese language, Portugal (pt-pt).
     */
    VerifyLanguages["PORTUGUESE_PORTUGAL"] = "pt-pt";
    /**
     * Romanian language (ro-ro).
     */
    VerifyLanguages["ROMANIAN"] = "ro-ro";
    /**
     * Russian language (ru-ru).
     */
    VerifyLanguages["RUSSIAN"] = "ru-ru";
    /**
     * Swedish language (sv-se).
     */
    VerifyLanguages["SWEDISH"] = "sv-se";
    /**
     * Thai language (th-th).
     */
    VerifyLanguages["THAI"] = "th-th";
    /**
     * Turkish language (tr-tr).
     */
    VerifyLanguages["TURKISH"] = "tr-tr";
    /**
     * Vietnamese language (vi-vn).
     */
    VerifyLanguages["VIETNAMESE"] = "vi-vn";
    /**
     * Yue Chinese Cantonese language (yue-cn).
     */
    VerifyLanguages["YUE_CHINESE_CANTONESE"] = "yue-cn";
    /**
     * Chinese language, Mainland (zh-cn).
     */
    VerifyLanguages["CHINESE_MAINLAND"] = "zh-cn";
    /**
     * Chinese language, Taiwan (zh-tw).
     */
    VerifyLanguages["CHINESE_TAIWAN"] = "zh-tw";
})(VerifyLanguages || (exports.VerifyLanguages = VerifyLanguages = {}));
