// @vonage/verify/dist/enums/VerifyStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifyStatus = void 0;
/**
 * Enum representing the status codes for Verify operations.
 */
var VerifyStatus;
(function (VerifyStatus) {
    /**
     * Success: Your user entered a correct verification code.
     */
    VerifyStatus["SUCCESS"] = "0";
    /**
     * Throttled: You have exceeded the rate limit for Verify requests.
     */
    VerifyStatus["THROTTLED"] = "1";
    /**
     * Missing Required Parameter: A required parameter is missing in the request.
     */
    VerifyStatus["MISSING_REQUIRED_PARAM"] = "2";
    /**
     * Invalid Parameter: An invalid parameter was provided in the request.
     */
    VerifyStatus["INVALID_PARAM"] = "3";
    /**
     * Invalid Credentials: The provided API credentials are invalid.
     */
    VerifyStatus["INVALID_CREDENTIALS"] = "4";
    /**
     * Internal Error: An internal error occurred during processing.
     */
    VerifyStatus["INTERNAL_ERROR"] = "5";
    /**
     * Failed to Process: The request failed to process.
     */
    VerifyStatus["FAILED_TO_PROCESS"] = "6";
    /**
     * Barred API Key: The API key used in the request is barred.
     */
    VerifyStatus["BARRED_API_KEY"] = "8";
    /**
     * Partner Quota Exceeded: The partner's quota for Verify requests has
     * been exceeded.
     */
    VerifyStatus["PARTNER_QUOTA_EXCEEDED"] = "9";
    /**
     * Concurrent Verifications: Multiple verifications are being processed
     * concurrently.
     */
    VerifyStatus["CONCURRENT_VERIFICATIONS"] = "10";
    /**
     * Unsupported Network: The network is not supported for verification.
     */
    VerifyStatus["UNSUPPORTED_NETWORK"] = "15";
    /**
     * Code Mismatch: The verification code entered by the user does not match.
     */
    VerifyStatus["CODE_MISMATCH"] = "16";
    /**
     * Invalid Code: The verification code provided is invalid.
     */
    VerifyStatus["INVALID_CODE"] = "17";
    /**
     * Delivery Failed: The delivery of the verification code failed.
     */
    VerifyStatus["DELIVERY_FAILED"] = "19";
    /**
     * PIN Not Supported: PIN verification is not supported.
     */
    VerifyStatus["PIN_NOT_SUPPORTED"] = "20";
    /**
     * Non-Permitted Destination: The destination is not permitted for
     * verification.
     */
    VerifyStatus["NON_PERMITTED_DESTINATION"] = "29";
})(VerifyStatus || (exports.VerifyStatus = VerifyStatus = {}));
