// @vonage/messages/dist/classes/WhatsApp/Audio.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Audio = void 0;
const WhatsAppAudio_1 = require("./WhatsAppAudio");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:whatsapp');
/**
 * @deprecated please use the WhatsAppAudio class instead
 *
 * @group WhatsApp
 */
class Audio extends WhatsAppAudio_1.WhatsAppAudio {
    /**
     * Constructs a new `Audio` instance for WhatsApp.
     *
     * @param {AudioObject} audio - The audio content of the message.
     * @param {string} to - The recipient's WhatsApp number.
     * @param {string} from - The sender's WhatsApp number.
     * @param {string} clientRef - (Optional) A unique client reference for the message.
     */
    constructor(audio, to, from, clientRef) {
        log('Please update to use the WhatsAppAudio class instead');
        super({
            audio: audio,
            to: to,
            from: from,
            clientRef: clientRef,
        });
    }
}
exports.Audio = Audio;
