// @vonage/messages/dist/classes/WhatsApp/File.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.File = void 0;
const WhatsAppFile_1 = require("./WhatsAppFile");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:whatsapp');
/**
 * @deprecated please use the WhatsAppFile class instead
 *
 * @group WhatsApp
 */
class File extends WhatsAppFile_1.WhatsAppFile {
    /**
     * Constructs a new `File` instance for WhatsApp.
     *
     * @param {FileObject} file - The file object for the message.
     * @param {string} to - The recipient's WhatsApp number.
     * @param {string} from - The sender's WhatsApp number.
     * @param {string} clientRef - (Optional) A unique client reference for the message.
     */
    constructor(file, to, from, clientRef) {
        log('Please update to use the WhatsAppFile class instead');
        super({
            to: to,
            from: from,
            clientRef: clientRef,
            file: file,
        });
    }
}
exports.File = File;
