// @vonage/messages/dist/classes/WhatsApp/Image.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Image = void 0;
const WhatsAppImage_1 = require("./WhatsAppImage");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:whatsapp');
/**
 * @deprecated Please use the WhatsAppImage class instead
 *
 * @group WhatsApp
 */
class Image extends WhatsAppImage_1.WhatsAppImage {
    /**
     * Constructs a new `Image` instance for WhatsApp.
     *
     * @param {ImageObject} image - The image object for the message.
     * @param {string} to - The recipient's WhatsApp number.
     * @param {string} from - The sender's WhatsApp number.
     * @param {string} clientRef - (Optional) A unique client reference for the message.
     */
    constructor(image, to, from, clientRef) {
        log('Please update to use the WhatsAppImage class instead');
        super({
            to: to,
            from: from,
            clientRef: clientRef,
            image: image,
        });
    }
}
exports.Image = Image;
