// @vonage/messages/dist/classes/WhatsApp/TemplateMessage.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplateMessage = void 0;
const WhatsAppTemplate_1 = require("./WhatsAppTemplate");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:whatsapp');
/**
 * @deprecated please use the WhatsAppTemplate class instead
 *
 * @group WhatsApp
 */
class TemplateMessage extends WhatsAppTemplate_1.WhatsAppTemplate {
    /**
     * Constructs a new `TemplateMessage` instance for WhatsApp.
     *
     * @param {MessageTemplate} template - The template object for the message.
     * @param {string} to - The recipient's WhatsApp number.
     * @param {string} from - The sender's WhatsApp number.
     * @param {string} locale - The locale or language code for the message.
     * @param {string} clientRef - (Optional) A unique client reference for the message.
     */
    constructor(template, to, from, locale, clientRef) {
        log('Please update to use the WhatsAppTemplate class instead');
        super({
            to: to,
            from: from,
            clientRef: clientRef,
            template: template,
            whatsapp: {
                policy: 'deterministic',
                locale: locale,
            },
        });
    }
}
exports.TemplateMessage = TemplateMessage;
