// @vonage/messages/dist/classes/WhatsApp/Text.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = void 0;
const WhatsAppText_1 = require("./WhatsAppText");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:whatsapp');
/**
 * @deprecated please use the WhatsAppText class instead
 *
 * @group WhatsApp
 */
class Text extends WhatsAppText_1.WhatsAppText {
    /**
     * Constructs a new `Text` instance for WhatsApp.
     *
     * @param {string} text - The text message content.
     * @param {string} to - The recipient's WhatsApp number.
     * @param {string} from - The sender's WhatsApp number.
     * @param {string} clientRef - (Optional) A unique client reference for the message.
     */
    constructor(text, to, from, clientRef) {
        log('Please update to use the WhatsAppText class instead');
        super({
            text: text,
            to: to,
            from: from,
            clientRef: clientRef,
        });
    }
}
exports.Text = Text;
