// @vonage/messages/dist/classes/WhatsApp/Video.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Video = void 0;
const WhatsAppVideo_1 = require("./WhatsAppVideo");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:whatsapp');
/**
 * @deprecated please use the WhatsAppVideo class instead
 *
 * @group WhatsApp
 */
class Video extends WhatsAppVideo_1.WhatsAppVideo {
    /**
     * Constructs a new `Video` instance for WhatsApp.
     *
     * @param {VideoObject} video - The video message content.
     * @param {string} to - The recipient's WhatsApp number.
     * @param {string} from - The sender's WhatsApp number.
     * @param {string} clientRef - (Optional) A unique client reference for the message.
     */
    constructor(video, to, from, clientRef) {
        log('Please update to use the WhatsAppVideo class instead');
        super({
            to: to,
            from: from,
            clientRef: clientRef,
            video: video,
        });
    }
}
exports.Video = Video;
