// @vonage/messages/dist/classes/WhatsApp/WhatsAppTemplate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WhatsAppTemplate = void 0;
const AbstractMessage_1 = require("../AbstractMessage");
/**
 * Represents a template message for WhatsApp.
 *
 * @group WhatsApp
 */
class WhatsAppTemplate extends AbstractMessage_1.AbstractMessage {
    /**
     * Send a template message to a WhatsApp user.
     *
     * @param {WhatsAppTemplateParams} params - The parameters for creating a WhatsApp template message.
     *
     * @example
     * ```ts
     * import { WhatsAppTemplate, WhatsAppLanguageCode } from '@vonage/messages';
     *
     * const { messageUUID } = await messagesClient.send(new WhatsAppTemplate({
     *  to: TO_NUMBER,
     *  from: FROM_NUMBER,
     *  whatsapp: {
     *    policy: 'deterministic',
     *    locale: WhatsAppLanguageCode.EN,
     *  },
     *  template: {
     *    name: 'your-template-name',
     *    parameters: [
     *      'foo',
     *      'bar',
     *    ],
     *  },
     *  clientRef: 'my-personal-reference',
     * }));
     *
     * console.log(`Message sent successfully with UUID ${messageUUID}`);
     * ```
     */
    constructor(params) {
        super(params);
        this.whatsapp = params.whatsapp;
        this.template = params.template;
        this.channel = 'whatsapp';
        this.messageType = 'template';
        if (params.context) {
            this.context = params.context;
        }
    }
}
exports.WhatsAppTemplate = WhatsAppTemplate;
