// @vonage/messages/dist/enums/Messenger/MessengerTags.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessengerTags = void 0;
/**
 * Enum representing message tags for Facebook Messenger messages.
 *
 * These tags are used to categorize the purpose and relevance of the message.
 *
 * @group Messenger
 */
var MessengerTags;
(function (MessengerTags) {
    /**
     * Represents a confirmed event update message tag.
     */
    MessengerTags["CONFIRMED_EVENT_UPDATE"] = "CONFIRMED_EVENT_UPDATE";
    /**
     * Represents a post-purchase update message tag.
     */
    MessengerTags["POST_PURCHASE_UPDATE"] = "POST_PURCHASE_UPDATE";
    /**
     * Represents an account update message tag.
     */
    MessengerTags["ACCOUNT_UPDATE"] = "ACCOUNT_UPDATE";
    /**
     * Represents a message tag for human agent communication.
     */
    MessengerTags["HUMAN_AGENT"] = "HUMAN_AGENT";
    /**
     * Represents a shipping update message tag.
     */
    MessengerTags["SHIPPING_UPDATE"] = "SHIPPING_UPDATE";
    /**
     * Represents a reservation update message tag.
     */
    MessengerTags["RESERVATION_UPDATE"] = "RESERVATION_UPDATE";
    /**
     * Represents a message tag for issue resolution.
     */
    MessengerTags["ISSUE_RESOLUTION"] = "ISSUE_RESOLUTION";
    /**
     * Represents an appointment update message tag.
     */
    MessengerTags["APPOINTMENT_UPDATE"] = "APPOINTMENT_UPDATE";
    /**
     * Represents a game event message tag.
     */
    MessengerTags["GAME_EVENT"] = "GAME_EVENT";
    /**
     * Represents a transportation update message tag.
     */
    MessengerTags["TRANSPORTATION_UPDATE"] = "TRANSPORTATION_UPDATE";
    /**
     * Represents a feature functionality update message tag.
     */
    MessengerTags["FEATURE_FUNCTIONALITY_UPDATE"] = "FEATURE_FUNCTIONALITY_UPDATE";
    /**
     * Represents a ticket update message tag.
     */
    MessengerTags["TICKET_UPDATE"] = "TICKET_UPDATE";
    /**
     * Represents a payment update message tag.
     */
    MessengerTags["PAYMENT_UPDATE"] = "PAYMENT_UPDATE";
    /**
     * Represents a personal finance update message tag.
     */
    MessengerTags["PERSONAL_FINANCE_UPDATE"] = "PERSONAL_FINANCE_UPDATE";
})(MessengerTags || (exports.MessengerTags = MessengerTags = {}));
