// @vonage/messages/dist/classes/Messenger/Audio.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Audio = void 0;
const MessengerAudio_1 = require("./MessengerAudio");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:messenger');
/**
 * @deprecated please use the MessengerAudio class instead
 *
 * @group Messenger
 */
class Audio extends MessengerAudio_1.MessengerAudio {
    /**
     * @deprecated Please use MessengerAudio instead.
     * Represents an audio message for the Messenger platform.
     *
     * @param {AudioObject} audio - The audio object containing the audio URL and an optional caption.
     * @param {string} to - The recipient's ID.
     * @param {string} from - The sender's ID.
     * @param {MessageType} messenger - The Messenger message type and category.
     * @param {string} clientRef - An optional client reference.
     */
    constructor(audio, to, from, messenger, clientRef) {
        log('Please update to use the MessengerAudio class instead');
        super({
            audio: audio,
            to: to,
            from: from,
            messenger: {
                category: messenger?.category,
                tag: messenger?.tag,
            },
            clientRef: clientRef,
        });
    }
}
exports.Audio = Audio;
