// @vonage/messages/dist/classes/Messenger/Text.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = void 0;
const MessengerText_1 = require("./MessengerText");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:messenger');
/**
 * Represents a text message for the Messenger channel.
 *
 * @deprecated Please use the MessengerText Class instead.
 *
 * @group Messenger
 */
class Text extends MessengerText_1.MessengerText {
    /**
     * Constructs a new `Text` instance.
     *
     * @param {string} text - The text content of the message.
     * @param {string} to - The recipient of the message.
     * @param {string} from - The sender of the message.
     * @param {MessageType} messenger - The messenger information for the message.
     * @param {string} clientRef - The client reference for the message.
     */
    constructor(text, to, from, messenger, clientRef) {
        log('Please update to use the MessengerText class instead');
        super({
            text: text,
            to: to,
            from: from,
            messenger: {
                category: messenger?.category,
                tag: messenger?.tag,
            },
            clientRef: clientRef,
        });
    }
}
exports.Text = Text;
