// @vonage/applications/dist/enums/VoiceRegions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceRegions = void 0;
/**
 * Enumeration representing different voice regions.
 *
 * @remarks
 * Selecting a region means all inbound, programmable SIP and SIP connect calls
 * will be sent to the selected region unless the call is sent to a regional endpoint.
 * If the call is using a regional endpoint this will override the application setting.
 */
var VoiceRegions;
(function (VoiceRegions) {
    /**
     * @description North America - East region
     */
    VoiceRegions["NA_EAST"] = "na-east";
    /**
     * @description North America - West region
     */
    VoiceRegions["NA_WEST"] = "na-west";
    /**
     * @description Europe - West region
     */
    VoiceRegions["EU_WEST"] = "eu-west";
    /**
     * @description Europe - East region
     */
    VoiceRegions["EU_EAST"] = "eu-east";
    /**
     * @description Asia-Pacific - Singapore region
     */
    VoiceRegions["APAC_SNG"] = "apac-sng";
    /**
     * @description Asia-Pacific - Australia region
     */
    VoiceRegions["APAC_AUSTRALIA"] = "apac-australia";
})(VoiceRegions || (exports.VoiceRegions = VoiceRegions = {}));
