// @vonage/sms/dist/enums/SMSErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SMSErrors = void 0;
/**
 * Enumeration representing possible SMS status codes.
 *
 * Defines different status codes that can be associated with SMS messages.
 *
 * @remarks This matches SMSStatus. Since Enums can't be joined, this is here
 * to help check valid statuses
 */
var SMSErrors;
(function (SMSErrors) {
    /**
     * Success: The message was sent successfully.
     */
    SMSErrors["SUCCESS"] = "0";
    /**
     * Throttled: Sending SMS faster than the account limit.
     */
    SMSErrors["THROTTLED"] = "1";
    /**
     * Missing Required Parameter: One or more required parameters are missing.
     */
    SMSErrors["MISSING_REQUIRED_PARAM"] = "2";
    /**
     * Invalid Parameter: The value of one or more parameters is invalid.
     */
    SMSErrors["INVALID_PARAM"] = "3";
    /**
     * Invalid Credentials: API key and/or secret are incorrect, invalid, or disabled.
     */
    SMSErrors["INVALID_CREDENTIALS"] = "4";
    /**
     * Internal Error: An error occurred while processing the message.
     */
    SMSErrors["INTERNAL_ERROR"] = "5";
    /**
     * Invalid Message: The platform was unable to process the message.
     */
    SMSErrors["INVALID_MESSAGE"] = "6";
    /**
     * Barred Number: The number you're trying to send messages to is blocklisted.
     */
    SMSErrors["BARRED_NUMBER"] = "7";
    /**
     * Partner Account Barred: Your Vonage account has been suspended.
     */
    SMSErrors["PARTNER_ACCOUNT_BARRED"] = "8";
    /**
     * Partner Quota Violation: Insufficient credit to send the message.
     */
    SMSErrors["PARTNER_QUOTA_VIOLATION"] = "9";
    /**
     * Too Many Existing Binds: Number of simultaneous connections exceeds account allocation.
     */
    SMSErrors["TOO_MANY_EXISTING_BINDS"] = "10";
    /**
     * Account Not Enabled For HTTP: Account is not provisioned for the SMS API.
     */
    SMSErrors["ACCOUNT_NOT_ENABLED"] = "11";
    /**
     * Message Too Long: Message length exceeds the maximum allowed.
     */
    SMSErrors["MESSAGE_TOO_LONG"] = "12";
    /**
     * Invalid Signature: The supplied signature could not be verified.
     */
    SMSErrors["INVALID_SIGNATURE"] = "14";
    /**
     * Invalid Sender Address: Using a non-authorized sender ID in the 'from' field.
     */
    SMSErrors["INVALID_SENDER_ADDRESS"] = "15";
    /**
     * Invalid Network Code: Network code is unrecognized or doesn't match the destination.
     */
    SMSErrors["INVALID_NETWORK_CODE"] = "22";
    /**
     * Invalid Callback URL: Callback URL is too long or contains illegal characters.
     */
    SMSErrors["INVALID_CALLBACK_URL"] = "23";
    /**
     * Non-Whitelisted Destination: Vonage account is in demo mode and requires whitelisted numbers.
     * This error is returned if you attempt to send a message to a number that isn't in your
     * pre-approved destination list.
     */
    SMSErrors["NON_WHITELISTED_DESTINATION"] = "29";
    /**
     * Signature And API Secret Disallowed: A signed request may not present an API secret.
     */
    SMSErrors["SIGNATURE_AND_API_SECRET_DISALLOWED"] = "32";
    /**
     * Number Deactivated: The number you're trying to send messages to is deactivated.
     */
    SMSErrors["NUMBER_DEACTIVATED"] = "33";
})(SMSErrors || (exports.SMSErrors = SMSErrors = {}));
