// @vonage/messages/dist/classes/MMS/Audio.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Audio = void 0;
const MMSAudio_1 = require("./MMSAudio");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:mms');
/**
 * Represents an audio message for the MMS channel.
 *
 * @deprecated Please use the MMSAudio class instead.
 *
 * @group MMS
 */
class Audio extends MMSAudio_1.MMSAudio {
    /**
     * Constructs a new `Audio` instance for the MMS channel.
     *
     * @param {AudioObject} audio - The audio content of the message.
     * @param {string} to - The recipient of the message.
     * @param {string} from - The sender of the message.
     * @param {string} clientRef - The client reference for the message.
     */
    constructor(audio, to, from, clientRef) {
        log('Please update to use the MMSAudio class instead');
        const params = {
            audio: audio,
            to: to,
            from: from,
            clientRef: clientRef,
        };
        super(params);
        this.channel = 'mms';
    }
}
exports.Audio = Audio;
