// @vonage/messages/dist/classes/MMS/Image.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Image = void 0;
const MMSImage_1 = require("./MMSImage");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:mms');
/**
 * Represents an image message for the MMS channel.
 *
 * @deprecated Please use the MMSImage class instead.
 *
 * @group MMS
 */
class Image extends MMSImage_1.MMSImage {
    /**
     * Constructs a new `Image` instance for the MMS channel.
     *
     * @param {ImageObject} image - The image content of the message.
     * @param {string} to - The recipient of the message.
     * @param {string} from - The sender of the message.
     * @param {string} clientRef - The client reference for the message.
     */
    constructor(image, to, from, clientRef) {
        log('Please update to use the MMSImage class instead');
        super({
            image: image,
            to: to,
            from: from,
            clientRef: clientRef,
        });
    }
}
exports.Image = Image;
