// @vonage/messages/dist/classes/MMS/Vcard.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vcard = void 0;
const MMSVcard_1 = require("./MMSVcard");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:mms');
/**
 * @deprecated Please use MMSVcard instead
 *
 * @group MMS
 */
class Vcard extends MMSVcard_1.MMSVcard {
    /**
     * Constructs a new `Vcard` instance for the MMS channel.
     *
     * @param {string} vcardUrl - The URL of the vCard.
     * @param {string} to - The recipient of the message.
     * @param {string} from - The sender of the message.
     * @param {string} clientRef - The client reference for the message.
     */
    constructor(vcardUrl, to, from, clientRef) {
        log('Please update to use the MMSVcard class instead');
        super({
            vcard: { url: vcardUrl },
            to: to,
            from: from,
            clientRef: clientRef,
        });
    }
}
exports.Vcard = Vcard;
