// @vonage/voice/dist/classes/NCCO/Notify.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Notify = void 0;
const enums_1 = require("../../enums");
/**
 * Represents a Notify action in an NCCO.
 */
class Notify {
    /**
     * Creates a new Notify action.
     *
     * @param {Object} payload - The payload data to send with the notification.
     * @param {string} eventUrl - The URL where the notification events will be sent.
     * @param {string} [eventMethod] - The HTTP method for sending notification events (e.g., "POST").
     */
    constructor(payload, eventUrl, eventMethod) {
        this.action = enums_1.NCCOActions.NOTIFY;
        this.payload = payload;
        this.eventUrl = [eventUrl];
        if (eventMethod) {
            this.eventMethod = eventMethod;
        }
    }
    /**
     * Serialize the Notify action to an NCCO-compatible format.
     *
     * @return {NotifyAction} - The serialized Notify action.
     */
    serializeToNCCO() {
        const data = {
            action: enums_1.NCCOActions.NOTIFY,
            payload: this.payload,
            eventUrl: this.eventUrl,
        };
        if (this.eventMethod) {
            data.eventMethod = this.eventMethod;
        }
        return data;
    }
}
exports.Notify = Notify;
