// @vonage/voice/dist/classes/Endpoint/WebsocketEndpoint.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebsocketEndpoint = void 0;
const debug_1 = __importDefault(require("debug"));
(0, debug_1.default)('@vonage/voice')(
// eslint-disable-next-line max-len
'This class is deprecated. Please update to use the WebsocketEndpointType type instead');
/**
 * Represents a WebSocket endpoint for making voice calls.
 *
 * @deprecated This class is deprecated. Please update to use the WebsocketEndpointType type instead.
 */
class WebsocketEndpoint {
    /**
     * Create a new WebsocketEndpoint instance.
     *
     * @param {string} uri - The WebSocket URI for the WebSocket endpoint.
     * @param {WebsocketBitrate} contentType - The content type or bitrate for WebSocket streaming.
     * @param {Record<string, unknown>} headers - Optional custom headers as key-value pairs.
     */
    constructor(uri, contentType, headers) {
        this.type = 'websocket';
        this.uri = uri;
        this.contentType = contentType;
        if (headers) {
            this.headers = headers;
        }
    }
}
exports.WebsocketEndpoint = WebsocketEndpoint;
