// @vonage/number-insights/dist/enums/NetworkType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkType = void 0;
/**
 * Enum representing the type of network associated with a phone number.
 */
var NetworkType;
(function (NetworkType) {
    /**
     * The phone number is associated with a mobile network.
     */
    NetworkType["MOBILE"] = "mobile";
    /**
     * The phone number is associated with a landline network.
     */
    NetworkType["LANDLINE"] = "landline";
    /**
     * The phone number is associated with a premium landline network.
     */
    NetworkType["LANDLINE_PREMIUM"] = "landline_premium";
    /**
     * The phone number is associated with a toll-free landline network.
     */
    NetworkType["LANDLINE_TOLLFREE"] = "landline_tollfree";
    /**
     * The phone number is associated with a virtual network.
     */
    NetworkType["VIRTUAL"] = "virtual";
    /**
     * The type of network associated with the phone number is unknown.
     */
    NetworkType["UNKNOWN"] = "unknown";
    /**
     * The phone number is associated with a pager network.
     */
    NetworkType["PAGER"] = "pager";
})(NetworkType || (exports.NetworkType = NetworkType = {}));
