// @vonage/messages/dist/classes/Viber/Image.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Image = void 0;
const ViberImage_1 = require("./ViberImage");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:viber');
/**
 * Represents an image message for the Viber channel.
 *
 * @deprecated Please use the ViberImage class instead.
 *
 * @group Viber
 */
class Image extends ViberImage_1.ViberImage {
    /**
     * Constructs a new `Image` instance for the Viber channel.
     *
     * @param {ImageObject} image - The image content of the message.
     * @param {string} to - The recipient's Viber ID.
     * @param {string} from - The sender's Viber ID.
     * @param {MessageConfig} viberService - The Viber service configuration.
     * @param {string} clientRef - The client reference for the message.
     */
    constructor(image, to, from, viberService, clientRef) {
        log('Please update to use the ViberImage class instead');
        super({
            image: image,
            to: to,
            from: from,
            clientRef: clientRef,
            viberService: viberService,
        });
    }
}
exports.Image = Image;
