// @vonage/messages/dist/classes/Viber/Text.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = void 0;
const ViberText_1 = require("./ViberText");
const debug_1 = __importDefault(require("debug"));
const log = (0, debug_1.default)('vonage:messages:viber');
/**
 * Represents a text message for the Viber channel.
 *
 * @deprecated Please use the ViberText class instead.
 *
 * @group Viber
 */
class Text extends ViberText_1.ViberText {
    /**
     * Constructs a new `Text` instance for the Viber channel.
     *
     * @param {string} text - The text content of the message.
     * @param {string} to - The recipient's Viber ID.
     * @param {string} from - The sender's Viber ID.
     * @param {MessageConfig} viberService - The Viber service configuration.
     * @param {string} clientRef - The client reference for the message.
     */
    constructor(text, to, from, viberService, clientRef) {
        log('Please update to use the ViberText class instead');
        super({
            text: text,
            to: to,
            from: from,
            viberService: viberService,
            clientRef: clientRef,
        });
    }
}
exports.Text = Text;
