// @vonage/verify2/dist/enums/channels.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SilentAuthChannel = exports.Channels = void 0;
/**
 * Enum representing different communication channels for verification.
 */
var Channels;
(function (Channels) {
    /**
     * SMS channel for verification.
     */
    Channels["SMS"] = "sms";
    /**
     * WhatsApp channel for verification.
     */
    Channels["WHATSAPP"] = "whatsapp";
    /**
     * Interactive WhatsApp channel for verification.
     */
    Channels["WHATSAPP_INTERACTIVE"] = "whatsapp_interactive";
    /**
     * Voice channel for verification.
     */
    Channels["VOICE"] = "voice";
    /**
     * Email channel for verification.
     */
    Channels["EMAIL"] = "email";
})(Channels || (exports.Channels = Channels = {}));
/**
 * Enum representing the Silent Authentication channel.
 */
var SilentAuthChannel;
(function (SilentAuthChannel) {
    /**
     * Silent Authentication channel for verification.
     */
    SilentAuthChannel["SILENT_AUTH"] = "silent_auth";
})(SilentAuthChannel || (exports.SilentAuthChannel = SilentAuthChannel = {}));
