// @vonage/verify2/dist/enums/verifyLocale.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifyLocale = void 0;
/**
 * Enum representing different locales for verification.
 */
var VerifyLocale;
(function (VerifyLocale) {
    /**
     * English (United States) locale.
     */
    VerifyLocale["EN_US"] = "en-us";
    /**
     * English (United Kingdom) locale.
     */
    VerifyLocale["EN_GB"] = "en-gb";
    /**
     * Spanish (Spain) locale.
     */
    VerifyLocale["ES_ES"] = "es-es";
    /**
     * Spanish (Mexico) locale.
     */
    VerifyLocale["ES_MX"] = "es-mx";
    /**
     * Spanish (United States) locale.
     */
    VerifyLocale["ES_US"] = "es-us";
    /**
     * Italian (Italy) locale.
     */
    VerifyLocale["IT_IT"] = "it-it";
    /**
     * French (France) locale.
     */
    VerifyLocale["FR_FR"] = "fr-fr";
    /**
     * German (Germany) locale.
     */
    VerifyLocale["DE_DE"] = "de-de";
    /**
     * Russian (Russia) locale.
     */
    VerifyLocale["RU_RU"] = "ru-ru";
    /**
     * Hindi (India) locale.
     */
    VerifyLocale["HI_IN"] = "hi-in";
    /**
     * Portuguese (Brazil) locale.
     */
    VerifyLocale["PT_BR"] = "pt-br";
    /**
     * Portuguese (Portugal) locale.
     */
    VerifyLocale["PT_PT"] = "pt-pt";
    /**
     * Indonesian (Indonesia) locale.
     */
    VerifyLocale["ID_ID"] = "id-id";
})(VerifyLocale || (exports.VerifyLocale = VerifyLocale = {}));
